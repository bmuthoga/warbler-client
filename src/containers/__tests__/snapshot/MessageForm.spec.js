import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import { MessageForm } from '../../MessageForm'

const props = {
  postNewMessage: jest.fn(),
  history: {
    push: jest.fn()
  },
  errors: {}
}

describe('MessageForm Snapshot Tests', () => {
  it('should render the MessageForm component without error message', () => {
    const wrapper = shallow(
      <MessageForm { ...props } />
    )

    expect(toJson(wrapper)).toMatchSnapshot()
  })
  
  it('should render the MessageForm component with error message', () => {
    const errors = {
      message: 'Sample error message'
    }
    const wrapper = shallow(
      <MessageForm { ...props } errors={errors} />
    )

    expect(toJson(wrapper)).toMatchSnapshot()
  })  
})
