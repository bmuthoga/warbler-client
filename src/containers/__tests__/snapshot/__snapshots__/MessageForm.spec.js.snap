// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageForm Snapshot Tests should render the MessageForm component with error message 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="alert alert-danger"
  >
    Sample error message
  </div>
  <input
    className="form-control"
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    className="btn btn-success pull-right"
    type="submit"
  >
    Add my message!
  </button>
</form>
`;

exports[`MessageForm Snapshot Tests should render the MessageForm component without error message 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    className="form-control"
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    className="btn btn-success pull-right"
    type="submit"
  >
    Add my message!
  </button>
</form>
`;
